### PR TITLE
Bumped the bundled Annoy to be 1.16.3 (not 1̶.̶1̶7̶.̶0̶)

### PR DESCRIPTION
**Pull request by mridang**
Based on my understanding anf git-foo, the 0.6.0 branch was built against 1.11.1 of @spotify/annoy. This can be seen by diffing https://github.com/spotify/annoy/blob/v1.11.1/src/annoylib.h and https://github.com/annoy4s/annoy4s/blob/v0.6.0/src/main/cpp/annoylib.h The `HammingWrapper` was cannibalised from another source file

### CODE REVIEW

File: src/test/scala/annoy4s/AnnoySpec.scala [hunk 1]
```diff
@@ -181,8 +181,8 @@ class AnnoySpec extends FlatSpec with Matchers {
 
     val index = Annoy.create[Int](tmpFile.pathAsString, numOfTrees = 2)
 
-    index.query(1, maxReturnSize = 10, searchK = 2).get.map(_._1) shouldBe List(1, 54, 55, 60, 76, 8, 32, 33)
-    index.query(1, maxReturnSize = 10, searchK = -1).get.map(_._1) shouldBe List(1, 69, 39, 87, 54, 29, 62, 55, 21, 35)
+    index.query(1, maxReturnSize = 10, searchK = 2).get.map(_._1) shouldBe List(1, 69, 39, 87, 54, 29, 62, 55, 21, 35)
```
Comment (mridang) [inline src/test/scala/annoy4s/AnnoySpec.scala]: My C++ foo is terrible. If @pishen could help verify this, that would be great. 🙏🏽

**Comment by pishen** [comment]
Thanks for the contribution, haven't come back for a long time, will need some time to pick it up 😅

**Comment by mridang** [comment]
The bug seems to be a (possible) gotcha in v1.17.0. I've opened an issue here: https://github.com/spotify/annoy/issues/540

**Comment by mridang** [comment]
@pishen all tests now pass so there isn't much for you to do except publish it. The version that is being used in now Annoy 1.16.3 since 1.17.0 is buggy. See my attached comment.

Any chance you could bump the version and release this, please? It should take only a few minutes. Thank you!

File: src/test/scala/annoy4s/AnnoySpec.scala (hunk 1 above)
Comment (mridang) [reply src/test/scala/annoy4s/AnnoySpec.scala]: Fixed. This is buggy in 1.17.0 and I've now moved to 1.16.3

**Comment by pishen** [comment]
Published as 0.10.0

**Pull request merged**